DPPP/DPInfo.cc

```diff
@@ -132,17 +132,17 @@ void DPInfo::set(std::vector<std::vector<double>>&& chan_freqs,
     }
   }
 
+  // Find the baseline with the most channels.
+  auto comp = [](const std::vector<double>& left,
+                 const std::vector<double>& right) {
+    return left.size() < right.size();
+  };
+  auto it = std::max_element(chan_freqs.begin(), chan_freqs.end(), comp);
```

Comment:
Although this is a nice change that works with the MSBDAReader (so you can put nchan = 0 in the init and let set() decide this), it will break the BDAAverager (as seen in some tests).

Because you overwrite the info's nchan, a DPBuffer's nchan will be different than the info's nchan and this exception will trigger in BDAAverager.cc:
```
 if (buffer.getData().shape() != expected_shape ||
      buffer.getFlags().shape() != expected_shape ||
      weights.shape() != expected_shape) {
    throw std::runtime_error("BDAAverager: Invalid buffer shape");
  }
```